README.md

```diff
@@ -1,137 +1,104 @@
+**RECOMMENDATION**: The CoreCLR repo has a dependencies.props file at its root and one of the properties inside is CoreClrPackageVersion. Picking that version of the shared framework helps to ensure your private bits are compatible with the other binaries that aren't being replaced.
 
-This will generate new crossgen/R2R images locally using the same shared framework version
+### Example 4: See the commands that will be used to run the benchmark without actually running it
 
-This step will also set some environment variables that affect the behavior of the subsequent commands. You'll see in the console output some of the information about the environment variables that were set.
+    .\RunBenchmark.ps1 -WhatIf
 
-This step assumes the latest version of ASP.NET and the shared framework. Use the `-AspNetVersion` and `-FrameworkVersion` parameters to override these.
+Outputs:
 
-### Step 4:
+     ***** Step 1 - Set Versions *******
+    cd F:\github\JitBench
+    .\AspNet-SetVersions.ps1
+    
+     ***** Step 2 - Setup Dotnet Runtime and SDK *******
+    .\DotNet-Install.ps1 -SharedRuntime -InstallDir .dotnet -Channel master -Version 2.0.0 -Architecture x64
+    .\DotNet-Install.ps1 -InstallDir .dotnet -Channel master -Version 2.0.0 -Architecture x64
+    
+     ***** Step 3 - Generate Store *******
+    cp ./CreateStore/bugfix_sdk_1682/Microsoft.NET.ComposeStore.targets .dotnet/sdk/2.0.0/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.ComposeStore.targets
+    .\AspNet-GenerateStore.ps1 -InstallDir .store -Architecture x64 -Runtime win7-x64
+    
+     ***** Step 4 - Restore MusicStore *******
+    cd src\MusicStore
+    dotnet restore
+    
+     ***** Step 5 - Publish MusicStore *******
+    rm -r bin\Release\netcoreapp2.0\publish
+    dotnet publish -c Release -f netcoreapp2.0 --manifest F:\github\JitBench\.store\x64\netcoreapp2.0\artifact.xml /p:MvcRazorCompileOnPublish=false
+    
+     ***** Step 6 - Run MusicStore *******
+    cd bin\Release\netcoreapp2.0\publish
+    dotnet MusicStore.dll
 
-Restore dependency packages
+IMPORTANT: .\AspNet-SetVersions.ps1 sets environment variables and writes files that the other commands rely on to work. If you start running the commands manually, make sure the environment is set up first.
 
-`cd src\MusicStore`
+### Example 5: Run the benchmark with Precompiled Views enabled
 
-`dotnet restore`
+    .\RunBenchmark.ps1 -PrecompiledViews
 
-You should see that all of the ASP.NET dependencies that get restored during this stage have the same version number as the output of step 3. 
+By default precompiled views are not used. See the View Compilation section below for more info.
 
-### Step 5: 
+### Example 6: Do all the benchmark preparation steps, but don't run MusicStore
 
-Build/publish MusicStore
+    .\RunBenchmark.ps1 -SetupOnly
 
-**Windows**
+This runs steps 1-5 in the -WhatIf output from above. You can now run dotnet MusicStore.dll manually (perhaps in a loop or with ETW collection turned on). 
```

Comment:
I think is is awesome! This is a huge functional upgrade on what this repo is capable of.